pyproject.toml

```diff
@@ -98,6 +98,7 @@ dev = [
   "fsspec",
   "greenlet",
   "hypothesis",
+  "pydantic[email]>=2.0",
```

Comment:
Are we happy doing this @litestar-org/maintainers? This means that we lock pydantic @ v2.4ish but suffer a lower version of piccolo than is current.

If good to go, I'll split this into a separate PR.

Reply:
hmm, maybe we can't: https://github.com/litestar-org/litestar/actions/runs/6507707828/job/17675629575?pr=2437#step:6:1

Reply:
Created https://github.com/litestar-org/litestar/pull/2438 to discuss this.